### ISSUE TITLE
ProgressIndicator, disabled ProgressSteps should have tabIndex -1

### ISSUE THREAD

**Issue by kuhanloh** [carbon-design-system/carbon#6089]
## Environment

MAC os 10.15.4
Browser: Chrome 81.0.4044.138, Firefox 76.0.1 
JAWS, Keyboard navigation

## Detailed description

**What version of the Carbon Design System are you using?**
carbon-components-react@7.11.3
carbon-components@10.11.2

**What did you expect to happen?**
Tabbing with a keyboard should skip a disabled ProgressStep

**What happened instead?**
ProgressSteps currently are assigned a tabIndex of 1 even when they are disabled. An invalid or disabled ProgressStep should have a tabIndex of -1. This can be quite annoying for keyboard and JAWS users as they have to tab through all the ProgressSteps before being able to interact with the actual content of the current ProgressStep.

**What WCAG 2.1 checkpoint does the issue violate?**
https://www.w3.org/TR/WCAG21/#focus-order
https://www.w3.org/TR/WCAG21/#character-key-shortcuts

## Steps to reproduce the issue

Use a keyboard and tab through ProgressStep

**Please create a reduced test case in CodeSandbox**
https://codesandbox.io/s/broken-bird-ykhkd?file=/src/index.js
https://ykhkd.csb.app/

## Additional information

- Screenshots or code
![image](https://user-images.githubusercontent.com/18057339/82046421-c14de080-96a8-11ea-907b-1b0a9a7bb932.png)

https://github.com/carbon-design-system/carbon/blob/4f0ba7cedf702885d6362ff231a89399297c7618/packages/react/src/components/ProgressIndicator/ProgressIndicator.js#L266
https://github.com/carbon-design-system/carbon/blob/4f0ba7cedf702885d6362ff231a89399297c7618/packages/react/src/components/ProgressIndicator/ProgressIndicator.js#L110

- Notes
